uaclient/security_status.py

```diff
@@ -129,23 +129,35 @@ def get_update_status(service_name: str, ua_info: Dict[str, Any]) -> str:
 def filter_security_updates(
     packages: List[apt_package.Package],
 ) -> DefaultDict[str, List[Tuple[apt_package.Version, str]]]:
-    """Filters a list of packages looking for available security updates.
+    """Filters a list of packages looking for available updates.
 
-    Checks if the package has a greater version available, and if the origin of
-    this version matches any of the series' security repositories.
+    All versions greater than the installed one are reported, based on where
+    it is provided, including ESM pockets, excluding backports.
     """
     result = defaultdict(list)
 
     for package in packages:
-        for version in package.versions:
-            if version > package.installed:
-                for origin in version.origins:
-                    service = get_origin_information_to_service_map().get(
-                        (origin.origin, origin.archive)
-                    )
-                    if service:
-                        result[service].append((version, origin.site))
-                        break  # No need to loop through all the origins
+        if package.is_installed:
+            for version in package.versions:
+                if version > package.installed:
+                    is_security = False
+                    for origin in version.origins:
+                        service = get_origin_information_to_service_map().get(
+                            (origin.origin, origin.archive)
+                        )
+                        if service:
+                            result[service].append((version, origin.site))
+                            is_security = True
+                            break  # No need to loop through all the origins
+                    # Also no need to report backports at least for now...
+                    expected_origin = version.origins[0]
+                    if (
+                        not is_security
+                        and "backports" not in expected_origin.archive
+                    ):
+                        result["standard-updates"].append(
+                            (version, version.origins[0].site)
+                        )
```

Comment:
Could this report the same _version_ twice? once for -security and once -updates (I don't think we want that but maybe I'm misremembering)

Reply:
It won't. Any version will be only listed once. Then... well:
- if it is in `-security` AND in `-updates` (which is a pretty common case) we report security - and that makes sense as it is indeed a security update.
- if it is in esm-infra AND esm-apps, then OH WELL OH WELL the first origin gets picked based on apt configuration, but this is a corner case for a very limited set of packages, and when we implemented this the first time we agreed that was good. Let me know if there is any major concern now about this decision.

Reply:
Ah! You're right - I misread the code. This all sounds good.